File: Day-1/cpp/part1.cpp
#include <fstream>
#include <vector>
#include <iostream>
#include <algorithm>
#include <cstdlib>

int main() {
    std::ifstream file("../data.txt");
    std::vector<int> data_first, data_second, indexes;
    int x, y, n;
    n = 0;
    while (file >> x >> y) {
        data_first.push_back(x);
        data_second.push_back(y);
        indexes.push_back(n);
        n++;
    }
    std::sort(data_first.begin(), data_first.end());
    std::sort(data_second.begin(), data_second.end());
    int total = 0;
    for(const int& n : indexes) {
        total += std::abs(data_first[n] - data_second[n]);
    }
    std::cout << total << std::endl;
}

File: Day-1/rust/src/part1.rs
// part1.rs

pub fn solve(input_path: &str) -> Result<i32, Box<dyn std::error::Error>> {
    let data_pairs: Vec<(i32, i32)> = std::fs::read_to_string(input_path)?
        .lines()
        .map(|line| {
            let mut nums = line.split("   ");
            (nums.next().unwrap().parse().unwrap(), 
             nums.next().unwrap().parse().unwrap())
        })
        .collect();
    let mut data_first: Vec<i32> = data_pairs.iter()
        .map(|(x,_)| *x)
        .collect::<Vec<_>>();
    let mut data_second: Vec<i32> = data_pairs.iter()
        .map(|(_,y)| *y)
        .collect::<Vec<_>>();
    
    data_first.sort();
    data_second.sort();
    let total: i32 = (0..data_first.len())
        .map(|i: usize| data_first[i] - data_second[i])
        .map(i32::abs)
        .sum();
    // Rest of solution here
    Ok(total) // Replace with actual result
}